Sources/Apollo/NetworkTransport.swift

```diff
@@ -8,8 +8,60 @@ public protocol NetworkTransport {
   ///   - completionHandler: A closure to call when a request completes. On `success` will contain the response received from the server. On `failure` will contain the error which occurred.
   /// - Returns: An object that can be used to cancel an in progress request.
   func send<Operation>(operation: Operation, completionHandler: @escaping (_ result: Result<GraphQLResponse<Operation>, Error>) -> Void) -> Cancellable
+  
+  /// The name of the client to send as the `"apollographql-client-name"` header.
+  var clientName: String { get set }
+  
+  /// The version of the client to send as the `"apollographql-client-version"` header
+  var clientVersion: String { get set }
 }
 
+public extension NetworkTransport {
+  
+  /// The header field name for the Client Name
+  static var headerFieldNameClientName: String {
+    return "apollographql-client-name"
+  }
+  
+  /// The header field name for the client version
+  static var headerFieldNameClientVersion: String {
+    return "apollographql-client-version"
+  }
+  
+  /// The default client name to use when setting up the `clientName` property
+  static var defaultClientName: String {
+    guard let identifier = Bundle.main.bundleIdentifier else {
+      return "apollo-ios-client"
+    }
+    
+    return "\(identifier)-apollo-ios"
+  }
+  
+  /// The default client version to use when setting up the `clientVersion` property.
+  static var defaultClientVersion: String {
+    var version = String()
+    if let shortVersion = Bundle.main.shortVersion {
+      version.append(shortVersion)
+    }
+    
+    if let buildNumber = Bundle.main.buildNumber {
+      if version.isEmpty {
+        version.append(buildNumber)
+      } else {
+        version.append("-\(buildNumber)")
```

Comment:
I actually decided against this on my end as I'm increasing the build number on each build (based of number of commits) and it makes the "Clients" view in Apollo Engine a bit noisy for the non-public environments/variants. YMMV of course and it's always configurable anyway.